tests/test_java.py

```diff
@@ -670,31 +672,60 @@ def test_java_different_basename(
         ) as container:
             application_pid = container.attrs["State"]["Pid"]
             profile = snapshot_pid_profile(profiler, application_pid)
-            if change_argv0:
-                # we changed basename - we should have run the profiler
-                assert profile.app_metadata is not None
-                assert (
-                    os.path.basename(profile.app_metadata["execfn"])
-                    == os.path.basename(profile.app_metadata["exe"])
-                    == java_notjava_basename
-                )
-                assert_function_in_collapsed(f"{java_notjava_basename};", profile.stacks)
-                assert_collapsed(profile.stacks)
-            else:
-                # we didn't change basename - we should not have run the profiler due to a different basename.
-                assert profile.stacks == Counter(
-                    {f"{java_notjava_basename};[Profiling skipped: profiling this JVM is not supported]": 1}
-                )
-                log_records = list(
-                    filter(
-                        lambda r: r.message
-                        == "Non-java basenamed process (cannot get Java version), skipping... (disable"
-                        " --java-safemode=java-extended-version-checks to profile it anyway)",
-                        caplog.records,
-                    )
-                )
-                assert len(log_records) == 1
-                assert os.path.basename(log_record_extra(log_records[0])["exe"]) == java_notjava_basename
+            assert profile.app_metadata is not None
+            assert (
+                os.path.basename(profile.app_metadata["execfn"])
+                == os.path.basename(profile.app_metadata["exe"])
+                == java_notjava_basename
+            )
+            assert_function_in_collapsed(f"{java_notjava_basename};", profile.stacks)
+            assert_collapsed(profile.stacks)
+
+
+@pytest.mark.parametrize("libjvm_removed", [False, True], ids=["libjvm_intact", "libjvm_removed"])
+def test_non_java_basename_version(
+    docker_client: DockerClient,
+    application_docker_image: Image,
+    assert_collapsed: AssertInCollapsed,
+    profiler_state: ProfilerState,
+    libjvm_removed: bool,
+) -> None:
+    """
+    Tests that we can profile and collect version for a java application with a different basename.
+    """
+    java_notjava_basename = "java-notjava"
+
+    with _application_docker_container(
+        docker_client,
+        application_docker_image,
+        [],
+        [],
+        application_docker_command=[
+            "bash",
+            "-c",
+            f"{java_notjava_basename} -jar Fibonacci.jar",
+        ],
+    ) as container:
+        logging.warning(container.logs())
+        application_pid = container.attrs["State"]["Pid"]
+        if libjvm_removed:
+            # test extracting java version even if libjvm was replaced
```

Comment:
Copied from `test_java_deleted_libjvm` right? Let's extract this to a tests util function `simulate_libjvm_delete`. Your code here misses the last `is_libjvm_deleted` assert btw, would be no issue if it was a shared function :)